src/irmin-pack/unix/pack_index.ml

```diff
@@ -39,9 +39,16 @@ module Make (K : Irmin.Hash.S) = struct
       Bytes.set_int64_be buf 0 (Int63.to_int64 off);
       Bytes.set_int32_be buf 8 (Int32.of_int len);
       Bytes.set buf 12 (Pack_value.Kind.to_magic kind);
-      Bytes.unsafe_to_string buf
+      (* Bytes.unsafe_to_string usage: buf is local, uniquely owned. We assume the various
+         functions above return unique ownership of buf. Then in the call to
+         Bytes.unsafe_to_string we give up unique ownership of buf for ownership of the
+         resulting string. This is safe. *)
+      Bytes.unsafe_to_string buf (* safe: see comment above *)
 
     let decode s pos : t =
+      (* Bytes.unsafe_of_string usage: s is shared ownership; buf is shared ownership (we
+         cannot mutate buf); we assume the Bytes.get... functions require only shared
+         ownership. This usage is safe. *)
```

Comment:
I view this as safe because

1. the buffer is never mutated (you cover this)
2. the lifetime of the buffer is this function

Do you think it is worth adding 2. to the comment (if I am looking at this correctly, of course)?